libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts

```diff
@@ -192,6 +194,8 @@ export class AdaptiveDialog<O extends object = {}> extends DialogContainer<O> {
     public async continueDialog(dc: DialogContext): Promise<DialogTurnResult> {
         this.onPushScopedServices(dc.context);
         try {
+            await this.checkForVersionChange(dc);
```

Comment:
>  await this.checkForVersionChange(dc); [](start = 10, length = 39)

I don't think this works. 
Doesn't every component have to expose version information?
Doesn't AdaptiveDialog version have to capture all of the stuff? 
```
            StringBuilder sb = new StringBuilder();
            sb.Append(base.GetVersion());  // this is the .Dialogs
            sb.Append(this.Id ?? string.Empty);
            sb.Append(this.AutoEndDialog);
            sb.Append(this.Recognizer.GetVersion());
            sb.Append(this.Selector.GetVersion());
            sb.Append(this.Generator.GetVersion());
            sb.Append(this.Schema.GetVersion());
            foreach (var trigger in Triggers)
            {
                sb.Append(trigger.GetVersion());  // Trigger itself is made up of conditions, and all of the nested ActionScopes
            }
```